src/menu/Menu.ts

```diff
@@ -0,0 +1,186 @@
+		this.own(createHandle(() => {
+			this._domNode = null;
+		}));
+	}
+
+	render(): DNode {
+		const {
+			id = this._id,
+			role = 'menu'
+		} = this.properties;
+
+		return v('div', {
+			classes: this.classes.apply(this, this._getMenuClasses()),
+			id,
+			key: 'root',
+			onfocusin: this._onMenuFocus,
+			onfocusout: this._onMenuFocusOut,
+			onkeydown: this._onMenuKeyDown,
+			role
+		}, this._renderChildren());
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		if (key === 'root') {
+			this._domNode = element;
+		}
+	}
+
+	private _getDefaultOrientation(): Orientation {
+		const { role = 'menu' } = this.properties;
+		return role === 'menubar' ? 'horizontal' : 'vertical';
+	}
+
+	private _getKeys() {
+		const { orientation = this._getDefaultOrientation() } = this.properties;
+		const isHorizontal = orientation === 'horizontal';
+
+		return {
+			decrease: isHorizontal ? Keys.left : Keys.up,
+			increase: isHorizontal ? Keys.right : Keys.down,
+			tab: Keys.tab
+		};
+	}
+
+	private _getMenuClasses() {
+		const { orientation = this._getDefaultOrientation() } = this.properties;
+		const classes = [ css.root ];
+
+		if (orientation === 'horizontal') {
+			classes.push(css.horizontal);
+		}
+
+		return classes;
+	}
+
+	private _isActive() {
+		return this._active || this.properties.active;
+	}
+
+	private _moveActiveIndex(operation: Operation) {
+		const max = this.children.length;
+		const previousIndex = this._activeIndex;
+		this._activeIndex = operation === Operation.decrease ?
+			previousIndex - 1 < 0 ? max - 1 : previousIndex - 1 :
+			Math.min(previousIndex + 1, max) % max;
+
+		this.invalidate();
+	}
+
+	private _onMenuFocus() {
+		if (!this._isActive()) {
+			this._active = true;
+			this.invalidate();
+		}
+	}
+
+	private _onMenuFocusOut() {
+		if (this._isActive()) {
+			requestAnimationFrame(() => {
+				const node = this._domNode as HTMLElement;
+				if (node && !node.contains(document.activeElement)) {
+					this._active = false;
+					this.invalidate();
+				}
+			});
+		}
+	}
+
+	private _onMenuKeyDown(event: KeyboardEvent) {
+		const keys = this._getKeys();
+
+		switch (event.keyCode) {
+			case keys.tab:
+				event.stopPropagation();
+				this._active = false;
+				this.invalidate();
+				break;
+			case keys.decrease:
+				event.preventDefault();
+				event.stopPropagation();
+				this._moveActiveIndex(Operation.decrease);
+				break;
+			case keys.increase:
+				event.preventDefault();
+				event.stopPropagation();
+				this._moveActiveIndex(Operation.increase);
+				break;
+		}
+	}
+
+	private _onMenuItemMouseDown(index?: number) {
+		if (typeof index === 'number') {
+			this._activeIndex = index;
+		}
+	}
+
+	private _renderChildren() {
+		const { activeIndex = this._activeIndex } = this.properties;
+		const focusIndex = this._isActive() ? activeIndex : undefined;
+		this._activeIndex = activeIndex;
+
+		this.children.filter((child: any) => child && child.properties)
```

Comment:
No, parents will need to render to a `<nav>` if needed since not every menu warrants a `<nav>` (e.g., dropdown menus within a parent `<nav>`).